### PR TITLE
Update campaigns ADDEDIT + Add media types

### PR DESCRIPTION
**Pull request by dziemba**

### CODE REVIEW

File: lib/soapy_cake/response.rb [hunk 1]
```diff
@@ -38,9 +38,23 @@ def sax
...
+    def error_check_special_case?
+      # Don't ask...
+      # As you might imagine, CAKE simply does not return the success element
+      # for this specific request. Also, this is the only request with a tag depth
+      # of 4, not 3 or 5 like ALL OTHER requests.
+      # BTW: There is a 10$ reward if anyone can find a worse designed API.
+      return true if sax.for_tag(:MediaType).count > 0
+
+      false
+    end
```
Comment (ndreynolds) [inline lib/soapy_cake/response.rb]: It's too bad we have to handle this case, but the approach seems pretty reasonable to me, given the CAKE API and all.

File: lib/soapy_cake/helper.rb [hunk 2]
```diff
@@ -41,5 +41,9 @@ def const_lookup(type, key)
         fail ArgumentError, "#{key} is not a valid value for #{type}"
       end
     end
+
+    def future_expiration_date
+      Date.today + (365 * 30)
```
Comment (ndreynolds) [inline lib/soapy_cake/helper.rb]: A little comment explaining the significance of that date would be nice to have.  Is it just an arbitrary date far into the future?

File: lib/soapy_cake/admin.rb [hunk 3]
```diff
@@ -102,6 +102,10 @@ def affiliate_signup(opts = {})
       run Request.new(:admin, :signup, :affiliate, opts)
     end
 
+    def media_types(*)
```
Comment (ndreynolds) [inline lib/soapy_cake/admin.rb]: Do we actually use the variable-length argument array anywhere?  If not, it seems it a little weird to me that we accept arguments and then just throw them away.  Just curious if there's a reason for that.

File: lib/soapy_cake/helper.rb (hunk 2 above)
Comment (dziemba) [inline lib/soapy_cake/helper.rb]: Added sarcastic comment.

File: lib/soapy_cake/admin.rb (hunk 3 above)
Comment (dziemba) [inline lib/soapy_cake/admin.rb]: No we don't. It's a leftover of the testing logic that passes a default argument of `{}` to all methods. I will fix that.

File: lib/soapy_cake/helper.rb [hunk 4]
```diff
@@ -41,5 +41,13 @@ def const_lookup(type, key)
         fail ArgumentError, "#{key} is not a valid value for #{type}"
       end
     end
+
+    # Some API calls require expiration dates.
+    # The default is to not expire campaigns/offers/etc., so we set this to far in the future.
+    # It cannot be *that* far in the future though because it causes a datetime overflow
+    # in the steam powered rusty black box they call a database server.
```
Comment (tsujigiri) [inline lib/soapy_cake/helper.rb]: :trollface:

**Comment by tsujigiri** [comment]
Looks good to me. (Except for the failing tests, of course.)

**Comment by ndreynolds** [comment]
:thumbsup: Looks good to me, too.

**Pull request merged**